lib/services/profiles-api-service.js

```diff
@@ -102,41 +102,80 @@ function profileApiServiceFactory(
 
                 });
             } else {
-                return self.createNodeAndRunDiscovery(macAddresses);
+                return self.createNodeAndRunDiscovery(macAddresses, options);
             }
         });
     };
 
-    ProfileApiService.prototype.runDiscovery = function(node) {
+    ProfileApiService.prototype.runDiscovery = function(node, options) {
         var self = this;
+        var configuration;
 
+        if (node.type === 'switch') {
+            configuration = self.getSwitchDiscoveryConfiguration(node, options.switchVendor);
+        } else {
+            configuration = {
+                name: 'Graph.SKU.Discovery',
+                options: {
+                    defaults: {
+                        graphOptions: {
+                            target: node.id
+                        },
+                        nodeId: node.id
+                    }
+                }
+            };
+        }
+
+        // The nested workflow holds the lock against the nodeId in this case,
+        // so don't add it as a target to the outer workflow context
+        return workflowApiService.createAndRunGraph(configuration, null)
+        .then(function() {
+            return self.waitForDiscoveryStart(node.id);
+        })
+        .then(function() {
+            return node;
+        });
+    };
+
+    ProfileApiService.prototype.getSwitchDiscoveryConfiguration = function(node, vendor) {
         var configuration = {
-            name: 'Graph.SKU.Discovery',
+            name: 'Graph.SKU.Switch.Discovery.Active',
             options: {
                 defaults: {
                     graphOptions: {
                         target: node.id
                     },
                     nodeId: node.id
+                },
+                'vendor-discovery-graph': {
+                    graphName: null
                 }
             }
         };
 
-        return workflowApiService.createAndRunGraph(configuration)
-        .then(function() {
-            return self.waitForDiscoveryStart(node.id);
-        })
-        .then(function() {
-            return node;
-        });
+        vendor = vendor.toLowerCase();
+
+        if (vendor === 'cisco') {
```

Comment:
@jlongever The downside of passing in the filename as a vendor option happens when the cataloging workflow gets more complicated. If we have three cataloging tasks, now we have to pass in three filenames for each of the cataloging scripts, but the workflow still has to label those tasks separately as what they are, mark the catalog format and source, etc. I think decoupling those values in this case would just make things harder, because now they are defined in two places instead of one.

Yep, the example graphs are probably better placed in on-skupack. I'll make a PR for that, maybe as a cisco-nexus-example SKU?